src/core/smallvec_type.hpp

```diff
@@ -7,14 +7,131 @@
  * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/** @file smallvec_type.hpp Simple vector class that allows allocating an item without the need to copy this->data needlessly. */
+/** @file smallvec_type.hpp Simple vector class that allows allocating an item without the need to copy this->data needlessly.
+ * @note The std::vector adapter functions are _temporary_ substitutions for methods of SmallVector
+ */
 
 #ifndef SMALLVEC_TYPE_HPP
 #define SMALLVEC_TYPE_HPP
 
 #include "alloc_func.hpp"
 #include "mem_func.hpp"
 
+#include <vector>
+#include <algorithm>
+
+/** Get a const iterator to a matching element in the vector.
+ * @param vec the vector to be searched
+ * @param elem the element to search fo
+ * @tparam T the contained type of the vector
+ */
+template <typename T>
+typename std::vector<T>::const_iterator Find(const std::vector<T>& vec, const T& elem) {
```

Comment:
opening brace of a function should be on a newline

(applies to various functions)